upf/upf_pfcp_api.c

```diff
@@ -535,14 +536,17 @@ handle_association_setup_request (pfcp_msg_t * req,
 
   SET_BIT (resp.grp.fields, ASSOCIATION_SETUP_RESPONSE_UP_FUNCTION_FEATURES);
   resp.up_function_features |= F_UPFF_EMPU;
-  /* currently no optional features are supported */
-
-  build_user_plane_ip_resource_information
-    (&resp.user_plane_ip_resource_information);
-  if (vec_len (resp.user_plane_ip_resource_information) != 0)
-    SET_BIT (resp.grp.fields,
-	     ASSOCIATION_SETUP_RESPONSE_USER_PLANE_IP_RESOURCE_INFORMATION);
-
+  if (gtm->pfcp_spec_version >= 16)
+  {
+    resp.up_function_features |= F_UPFF_FTUP;
+  } else
+  {
```

Comment:
VPP has very strict code formatting rules and we should IMHO follow those. The `else` is not following them.